src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -147,12 +162,14 @@ public void Dispose()
 
         private void Dispose(bool disposing)
         {
-            if (disposing)
+            if (disposing && !isDisposed)
```

Comment:
What would be the point of that?
Calling Dispose over and over again is supposed to be safe, part of the pattern is to check if it has already been called and ignore future calls.  Since the transmitter has already been disposed of as well it accomplishes nothing to call it again either.